alot/commands/__init__.py

```diff
@@ -136,15 +163,16 @@ def __init__(self, mode, name, help=None, usage=None,
         self.forced = forced or {}
         self.arguments = arguments or []
 
-    def __call__(self, klass):
-        helpstring = self.help or klass.__doc__
+    def __call__(self, class_):
+        helpstring = self.help or class_.__doc__
         argparser = CommandArgumentParser(description=helpstring,
                                           usage=self.usage,
                                           prog=self.name, add_help=False)
         for args, kwargs in self.arguments:
             argparser.add_argument(*args, **kwargs)
-        COMMANDS[self.mode][self.name] = (klass, argparser, self.forced)
-        return klass
+        COMMANDS[self.mode][self.name] = (class_, argparser, self.forced)
+        REVERSE_COMMANDS[class_] = (self.name, self.mode)
```

Comment:
How about turning `REVERSE_COMMANDS` into a dict `HOOKS`,
that maps `class_` to a pair `(prehook,posthook)`, and which gets populated here?
The `get_hook` function I suggested above is then just `HOOKS[class_]`.